app/tasks/launch.js

```diff
@@ -105,6 +114,46 @@ let self = {
     return self.launch(exe_path, [], opts)
   },
 
+  launch_html_cave: async function(opts, cave) {
+    let game = await db.find_one({_table: 'games', id: cave.game_id})
+    let app_path = CaveStore.app_path(cave.install_location, opts.id)
+    if (!fs.existsSync(path.join(app_path, 'index.html'))) {
+      throw new Error('html game missing index.html')
+    }
+    let win = new BrowserWindow({
+      title: game.title,
+      icon: './static/images/itchio-tray-x4.png',
+      width: 1280, height: 720,
+      center: true,
+      show: true,
+      'auto-hide-menu-bar': true,
+      'node-integration': false
+    })
+    let serve = serveStatic(app_path, {'index': ['index.html', 'index.htm']})
+    let server = http.createServer((req, res) => {
+      let done = finalhandler(req, res)
+      serve(req, res, done)
+    })
+    let port
+    server.listen(0) // let node/os assign random port
+    server.on('listening', function() {
+      port = server.address().port
+      log(opts, `serving game on port ${port}`)
+      win.loadURL(`http://localhost:${port}/index.html`)
+      if (process.env.DEVTOOLS === '1') {
+        win.webContents.openDevTools({detach: true})
+      }
+    })
+
+    return new Promise((resolve, reject) => {
```

Comment:
returning a promise is now an anti-pattern — instead, you can await one.

example:

```javascript
async function waste_time () {
// wait one second
  let p = new Promise((resolve, reject) => {
    setTimeout(resolve, 1000)
  })
  console.log(`Waiting one second...`)
  await p
  console.log(`Hey, there we are!`)  
}
```